### DISCUSSION TITLE
Multi-functional app

### DISCUSSION THREAD

**Discussion by shuzERR** [tui-cs/Terminal.Gui#2234] [Q&A]
Hello, I need to create app with login page and after logging in few features. I want to what is the best way to do this? I used sample code from Example.cs but modified 
`		btnLogin.Clicked += () => {
			if (usernameText.Text == "admin" && passwordText.Text == "password") {
				MessageBox.Query ("Logging In", "Login Successful", "Ok");
				Application.RequestStop ();
			} else {
				MessageBox.ErrorQuery ("Logging In", "Incorrect username or password", "Ok");
			}
		};` 
to 
`        btnLogin.Clicked += () => {
            if (usernameText.Text == "admin" && passwordText.Text == "password")
            {
                MessageBox.Query("Logging In", "Login Successful", "Ok");
                Application.RequestStop();
                Application.Run<MainWindow>();
            }
            else
            {
                MessageBox.ErrorQuery("Logging In", "Incorrect username or password", "Ok");
            }
        };`
 but now when I successfully log in, I still recieve MessageBox when clicking <enter>
Is that correct way or I should know abaout something else?

**Comment by tznind** [comment]
Your event handler for `btnLogin` has 3 statements 

```csharp
MessageBox.Query("Logging In", "Login Successful", "Ok"); 
Application.RequestStop(); 
Application.Run<MainWindow>(); 
```
> but now when I successfully log in, I still recieve MessageBox when clicking 

Do you mean that you still recieve the 'Successful' message? If so that is because you have are still invoking that method (`MessageBox.Query("Logging In", "Login Successful", "Ok"); `).

If you change your code to the following (remove the red bits from your code, add the green bits) it sounds like you will get what you want:

```diff
// When login button is clicked display a message popup
btnLogin.Clicked += () => {
	if (usernameText.Text == "admin" && passwordText.Text == "password") {
-		MessageBox.Query ("Logging In", "Login Successful", "Ok");
		Application.RequestStop ();
+		Application.Run<MainWindow> ();
	} else {
		MessageBox.ErrorQuery ("Logging In", "Incorrect username or password", "Ok");
	}
};
```
